activation/handler_v2.go

```diff
@@ -248,3 +306,319 @@ func (h *HandlerV2) collectAtxDeps(atx *wire.ActivationTxV2) ([]types.Hash32, []
 
 	return maps.Keys(poetRefs), maps.Keys(filtered)
 }
+
+func (h *HandlerV2) coinbase(ctx context.Context, atx *wire.ActivationTxV2) (types.Address, error) {
+	if atx.Coinbase != nil {
+		return *atx.Coinbase, nil
+	}
+	return atxs.Coinbase(h.cdb, atx.SmesherID)
+}
+
+func (h *HandlerV2) previous(ctx context.Context, id types.ATXID) (opaqueAtx, error) {
+	var blob sql.Blob
+	version, err := atxs.LoadBlob(ctx, h.cdb, id[:], &blob)
```

Comment:
Do we use fields other then those available in the `atxs` table for previous ATX validation?
That is, do we really need to decode and validate that blob?

Reply:
We need this because the ATX in the `atxs` table contains the _effective_ num units. Effective num units are the smallest value of the num units in the ATX and its previous ATX (the ones declared by the ATX itself). If we take a minimum of Atx.NumUnits and PrevATX.EffectiveNumUnits, then no one will be able to grow _effectively_ their PoST size.

Reply:
Interesting why then we're storing `atx.NumUnits` in the table. Do we have some kind of confusion here?
https://github.com/spacemeshos/go-spacemesh/blob/develop/sql/atxs/atxs.go#L380

Reply:
`NumUnits` as included by the ATX is only needed to be able to verify the provided post. Effective NumUnits (what we store as `NumUnits` in the DB) is used to calculate the actual weight of an identity. They are not the same.

To count your `NumUnits` on disk as effective numunits you need to hold them for at least one full poet round, i.e. you must generate 2 ATXs with that amount of storage - one before the poet round and one after. That's why effective `NumUnits` is `min(previous.NumUnits, atx.NumUnits)` (where both are the `wire` form of the ATX).